### ISSUE TITLE
TSQL: Comment parsing issues

### ISSUE THREAD

**Issue by jpers36** [sqlfluff/sqlfluff#2086]
I'm guessing this is related to #2027

## Expected Behaviour

Comments don't cause parsing issues.

## Observed Behaviour

Comments can throw parsing issues.

```
/**
**/

select 1
```

Throws:  Found unparsable section: '/**\n**/\n\nselect 1'

```
/**
(
**/

select 1
```

Throws:  Couldn't find closing bracket for opening bracket.

```
/**
'
**/

select 1
```

Throws: Fatal. Unable to lex characters: 0

## Steps to Reproduce

Run parser on above snippets

## Dialect

TSQL 

## Version

Latest main

**Comment by jpy-git** [comment]
@jpers36 Thanks for flagging this!

couple thoughts from initial tests:
- Still works for correctly for non-tsql. Therefore we can rule out the `regex` package causing any issues.
- can confirm that the following variations work in tsql:
```sql
/*
(
*/
```
or
```sql
/**
(
*/
```
(same for all those special characters)
- What is actually causing the issue is the closing brackets cause the pattern to not match:
i.e.
```sql
/*
(
**/
```
fails

**Comment by jpy-git** [comment]
I don't have access to T-SQL atm so could you just verify that a comment like `/* comment **/` is fine for me?

I expect it is, but useful to confirm 😄

**Comment by jpers36** [comment]
Yes, that's a perfectly acceptable comment.

**Comment by jpy-git** [comment]
awesome thanks will pull together a fix 👍
